src/platform/EFR32/CHIPCryptoPALPsaEfr32.cpp

```diff
@@ -0,0 +1,1939 @@
+/*
```

Comment:
Does it use any Sillicon Labs APIs or only mbedTLS/PSA crypto? I wonder if we could move it to `src/crypto` and reuse in other platforms easily as we were also planning working on that.

Reply:
It takes shortcuts by calling into the driver wrapper layer (to bypass having to import/delete a key in the PSA key database for every single operation), which is not declared a stable API. That's why IMO this implementation doesn't belong in `src/crypto` (and neither does the SE050 implementation, but that's another discussion).

You could easily build on it and make a generic mbedTLS/PSA implementation of the Matter Crypto HAL (just change out the `psa_driver_wrapper_xxx()` calls for their top-level equivalents and wrap them in a `psa_import_key()`/`psa_destroy_key()` pair), but it'd suffer from performance issues making it not worth it until Matter moves to abstract keys for the symmetric operations as well.

That is, unless you're trying to build for something like TF-M. Then I guess eating the overhead is your only option besides duplicating the symmetric crypto algorithm implementations in the non-secure world.

Reply:
A PSA API-compliant implementation has been prototyped here: https://github.com/stevew817/connectedhomeip/blob/5b1ad7f945ee7ffe67ee7633f79f7cdce20defa0/src/crypto/CHIPCryptoPALmbedTLS.cpp

Reply:
I see, thanks very much for the explanation. Yeah, I think we need to introduce a better abstraction for keys in Matter to be prepared for TF-M. :)